### ISSUE TITLE
Unable to `pootle collectstatic`

### ISSUE THREAD

**Issue by KINFOO** [translate/pootle#3443]
Hi,

I am installing _Pootle_ from e `Dockerfile`. Installation goes quite well, but when running `collectstatic` the drama starts:
```bash
Step 16 : RUN pootle collectstatic --noinput
 ---> Running in 0cffd1c57853
2015-01-05 11:05:56,407 INFO Loading custom settings from '/home/poo/.pootle/pootle.conf'...
2015-01-05 11:05:56,576 INFO Using Python PO                                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                   
  File "/home/poo/.local/bin/pootle", line 9, in <module>
    load_entry_point('Pootle==2.5.1.1', 'console_scripts', 'pootle')()
  File "/home/poo/.local/lib/python2.7/site-packages/pootle/runner.py", line 210, in main
    django_settings_module='pootle.settings')
  File "/home/poo/.local/lib/python2.7/site-packages/pootle/runner.py", line 198, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/management/__init__.py", line 443, in execute_from_command_line
    utility.execute()
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/management/__init__.py", line 382, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/management/base.py", line 196, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/management/base.py", line 232, in execute
    output = self.handle(*args, **options)
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/management/base.py", line 371, in handle
    return self.handle_noargs(**options)
  File "/home/poo/.local/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 163, in ha
    collected = self.collect()
  File "/home/poo/.local/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 104, in co
    for path, storage in finder.list(self.ignore_patterns):
  File "/home/poo/.local/lib/python2.7/site-packages/django_assets/finders.py", line 68, in list
    for output in super(AssetsFinder, self).list(ignore_patterns):
  File "/home/poo/.local/lib/python2.7/site-packages/django/contrib/staticfiles/finders.py", line 210, in list
    for path in utils.get_files(self.storage, ignore_patterns):
  File "/home/poo/.local/lib/python2.7/site-packages/django/contrib/staticfiles/utils.py", line 25, in get_files
    directories, files = storage.listdir(location)
  File "/home/poo/.local/lib/python2.7/site-packages/django/core/files/storage.py", line 234, in listdir
    for entry in os.listdir(path):
OSError: [Errno 2] No such file or directory: '/home/poo/.local/lib/python2.7/site-packages/pootle/assets'
```
How did I installed it?
```bash
$ cp /custom/conf ~/.pootle/pootle.conf
$ pip install --user -r $POOTLE_HOME/requirements
$ pootle setup
 ---> Running in f568e3f27bab
2015-01-05 11:03:31,149 INFO Loading custom settings from '/home/poo/.pootle/pootle.conf'...
2015-01-05 11:03:31,294 INFO Using Python PO                                                                                                                                                         
2015-01-05 11:03:31,390 INFO Setting up a new Pootle installation.                                                                                                                                   
Syncing...
Creating tables ...
Creating table django_session
Creating table auth_permission
Creating table auth_group_permissions
Creating table auth_group
Creating table auth_user_user_permissions
Creating table auth_user_groups
Creating table auth_user
Creating table django_content_type
Creating table django_site
Creating table django_admin_log
Creating table registration_registrationprofile
Creating table siteconfig_siteconfiguration
Creating table votes
Creating table south_migrationhistory
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.sessions
 > django.contrib.auth
 > django.contrib.contenttypes
 > django.contrib.sites
 > django.contrib.admin
 > django.contrib.messages
 > django.contrib.staticfiles
 > registration
 > djblets.siteconfig
 > django_assets
 > voting
 > south

Not synced (use migrations):
 - pootle_app
 - pootle_store
 - pootle_language
 - pootle_project
 - pootle_translationproject
 - pootle_profile
 - pootle_statistics
 - pootle_tagging
 - pootle_notifications
 - staticpages
 - taggit
(use ./manage.py migrate to migrate these)
Running migrations for pootle_app:
 - Migrating forwards to 0001_initial.
 > pootle_app:0001_initial
 - Loading initial data for pootle_app.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_store:
 - Migrating forwards to 0001_initial.
 > pootle_store:0001_initial
 - Loading initial data for pootle_store.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_language:
 - Migrating forwards to 0002_auto__del_field_language_description_html__chg_field_language_descript.
 > pootle_language:0001_initial
 > pootle_language:0002_auto__del_field_language_description_html__chg_field_language_descript
 - Loading initial data for pootle_language.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_project:
 - Migrating forwards to 0002_auto__del_field_project_description_html__chg_field_project_descriptio.
 > pootle_project:0001_initial
 > pootle_project:0002_auto__del_field_project_description_html__chg_field_project_descriptio
 - Loading initial data for pootle_project.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_translationproject:
 - Migrating forwards to 0002_auto__del_field_translationproject_description_html__chg_field_transla.
 > pootle_translationproject:0001_initial
 > pootle_translationproject:0002_auto__del_field_translationproject_description_html__chg_field_transla
 - Loading initial data for pootle_translationproject.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_profile:
 - Migrating forwards to 0001_initial.
 > pootle_profile:0001_initial
 - Loading initial data for pootle_profile.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_statistics:
 - Migrating forwards to 0001_initial.
 > pootle_statistics:0001_initial
 - Loading initial data for pootle_statistics.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_tagging:
 - Migrating forwards to 0001_initial.
 > pootle_tagging:0001_initial
 - Loading initial data for pootle_tagging.
Installed 0 object(s) from 0 fixture(s)
Running migrations for pootle_notifications:
 - Migrating forwards to 0001_initial.
 > pootle_notifications:0001_initial
 - Loading initial data for pootle_notifications.
Installed 0 object(s) from 0 fixture(s)
Running migrations for staticpages:
 - Migrating forwards to 0011_auto__chg_field_legalpage_modified_on__chg_field_staticpage_modified_o.
 > staticpages:0001_initial
 > staticpages:0002_rename_app
 > staticpages:0003_auto__del_field_legalpage_body_html__chg_field_legalpage_body
 > staticpages:0004_auto__add_field_legalpage_virtual_path
 > staticpages:0005_data__slug_to_virtual_path
 - Migration 'staticpages:0005_data__slug_to_virtual_path' is marked for no-dry-run.
 > staticpages:0006_auto__del_field_legalpage_slug__chg_field_legalpage_virtual_path
 > staticpages:0007_auto__add_unique_legalpage_virtual_path
 > staticpages:0008_auto__del_field_legalpage_display_on_register
/home/poo/.local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:808: RuntimeWarning: DateTimeField received a naive datetime (2015-01-05 11:05:01.992324) while time zone support is active.                                                                                                                                                                                             
  RuntimeWarning)                                                                                                                                                                                    
/home/poo/.local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:808: RuntimeWarning: DateTimeField received a naive datetime (2015-01-05 11:05:01.993565) while time zone support is active.                                                                                                                                                                                             
  RuntimeWarning)                                                                                                                                                                                    
/home/poo/.local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:808: RuntimeWarning: DateTimeField received a naive datetime (2015-01-05 11:05:02.753268) while time zone support is active.                                                                                                                                                                                             
  RuntimeWarning)                                                                                                                                                                                    
/home/poo/.local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:808: RuntimeWarning: DateTimeField received a naive datetime (2015-01-05 11:05:03.908433) while time zone support is active.                                                                                                                                                                                             
  RuntimeWarning)                                                                                                                                                                                    
2015-01-05 11:05:14,486 INFO Populating the database.                                                                                                                                                
2015-01-05 11:05:35,725 INFO Succesfully populated the database.                                                                                                                                     
2015-01-05 11:05:35,725 INFO Successfully deployed new Pootle.                                                                                                                                       
 > staticpages:0009_auto__add_staticpage
 > staticpages:0010_auto__add_agreement__add_unique_agreement_user_document__add_field_sta
 > staticpages:0011_auto__chg_field_legalpage_modified_on__chg_field_staticpage_modified_o
 - Loading initial data for staticpages.
Installed 0 object(s) from 0 fixture(s)
Running migrations for taggit:
 - Migrating forwards to 0002_unique_tagnames.
 > taggit:0001_initial
 > taggit:0002_unique_tagnames
 - Loading initial data for taggit.
Installed 0 object(s) from 0 fixture(s)
```
What is in `$POOTLE_HOME/requirements`.
```bash
$ cat $POOTLE_HOME/requirements
Pootle < 2.6
python-Levenshtein < 0.12
```
Post mortem analysis:
```bash
$ pip freeze
Warning: cannot find svn location for distribute==0.6.24dev-r0
Django==1.4.17
Pootle==2.5.1.1
South==0.7.6
argparse==1.2.1
cssmin==0.2.0
diff-match-patch==20121119
## FIXME: could not find svn URL in dependency_links for this package:
distribute==0.6.24dev-r0
django-assets==0.10
django-taggit==0.11.1
django-voting==0.1
lxml==3.4.1
python-Levenshtein==0.11.2
six==1.9.0
translate-toolkit==1.13.0-rc1
webassets==0.10
wsgiref==0.1.2
$ ls /home/poo/.local/lib/python2.7/site-packages/pootle
__init__.py      api_factory.py   dbs           locale      runner.py   settings.py          syspath_override.pyc
__init__.pyc     api_factory.pyc  depcheck.py   middleware  runner.pyc  settings.pyc         templates
__version__.py   apps             depcheck.pyc  po          scripts     static               urls.py
__version__.pyc  core             i18n          repos       settings    syspath_override.py  urls.pyc
```

How comes there is no `assets` in `site-packages/pootle`?

Regards
Kevin

**Comment by dwaynebailey** [comment]
Not sure what version you are trying this with.  But I think in 2.5.1.1 there was a packaging issue that didn't prepopulate the assets.  The solutions is to ``mkdir`` that missing ``assets`` directory.

**Comment by KINFOO** [comment]
Hi,

In _post mortem analysis_ section, `pip freeze` reveals `Pootle==2.5.1.1`.

Thanks for your quick answer, I am going to try this right away.
Kevin

**Comment by KINFOO** [comment]
@dwaynebailey It worked like a charm.

Does it worth working on fixing this for a future `2.5.1.2` or is it better to wait for `2.6.0`?
I am willing to help if needed.

Kevin

**Comment by unho** [comment]
@KINFOO We might release a quick 2.5.1.2 just to include this minor fix, but most likely we will release 2.7.0 instead. 2.6.0 will be released as well at the exactly same time as 2.7.0 to provide an upgrade path from older Pootle deployments.
